tests/phpunit/tests/formatting/wpCheckInvalidUtf8.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/**
+ * @group formatting
+ *
+ * @covers ::wp_check_invalid_utf8
+ */
+class Tests_Formatting_wpCheckInvalidUtf8 extends WP_UnitTestCase {
+
+	/**
+	 * Data provider for valid UTF-8 strings.
+	 *
+	 * @return array {
+	 *     @type array {
+	 *         @type string $value    The valid UTF-8 string.
+	 *     }
+	 * }
+	 */
+	public function data_valid_utf8_provider() {
```

Comment:
@Debarghya-Banerjee a few improvements:

1. Order: Put the data provider below the test method(s) using it.
2. DocBlock: there's not yet consensus on if a full detail of the `@return array` is needed. For consistency, please use only `@return array[]` without the expanded details. [Here's an example](https://core.trac.wordpress.org/browser/trunk/tests/phpunit/tests/functions/wpTriggerError.php#L81).
3. Debug: what gets reported might not clearly identify which dataset failed. Using named datasets (i.e. having a short description as a key) helps. For example, use the inline comments as the "key" for each array.

Reply:
Another improvement:

Standardizing on the following naming convention for data provider:

If it relates to only 1 test method, then use that test method's name except replace the `test_` prefix with `data_`.

```suggestion
	public function data_should_return_given_string_when_valid_without_stripping() {
```